libs/api/libbpf_program.cpp

```diff
@@ -422,3 +420,130 @@ bpf_prog_bind_map(int prog_fd, int map_fd, const struct bpf_prog_bind_opts* opts
 
     return libbpf_result_err(ebpf_program_bind_map(prog_fd, map_fd));
 }
+
+static int
+__bpf_set_link_xdp_fd_replace(int ifindex, int fd, int old_fd, __u32 flags)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+
+    // On Linux, the logic to detach the older program and attach the
+    // new program is present in kernel (the flag XDP_FLAGS_REPLACE is
+    // passed to the kernel).  Thus, we should consider moving this
+    // logic to the execution context.
+    if (flags & XDP_FLAGS_REPLACE) {
+        // Look up the old program info to get the program ID.
+        ebpf_id_t program_id = 0;
+        if (old_fd == 0 || old_fd == ebpf_fd_invalid) {
```

Comment:
Minor: should we fail the call if `XDP_FLAGS_REPLACE` is set and `old_fd` is `0` or `ebpf_fd_invalid`?
Linux behavior is not clear by looking at the libbpf code.

Reply:
Unless we find an app compat issue, I think it's fine to support more functionality than Linux rather than failing the call.